frmts/ogcapi/gdalogcapidataset.cpp

```diff
@@ -655,6 +658,44 @@ bool OGCAPIDataset::InitFromFile(GDALOpenInfo *poOpenInfo)
     return InitFromCollection(poOpenInfo, oDoc);
 }
 
+/************************************************************************/
+/*                        ProcessScale()                          */
+/************************************************************************/
+
+bool OGCAPIDataset::ProcessScale(CPLJSONObject &oScaleDenominator,
+                                 const double dfXMin, const double dfYMin,
+                                 const double dfXMax, const double dfYMax)
+
+{
+    bool bRet = FALSE;
+    double dfRes = 1e-8;  // arbitrary
+    if (oScaleDenominator.IsValid())
+    {
+        const double dfScaleDenominator = oScaleDenominator.ToDouble();
+        constexpr double HALF_CIRCUMFERENCE = 6378137 * M_PI;
+        dfRes = dfScaleDenominator / ((HALF_CIRCUMFERENCE / 180) / 0.28e-3);
+    }
+
+    double dfXSize = (dfXMax - dfXMin) / dfRes;
+    double dfYSize = (dfYMax - dfYMin) / dfRes;
+    while (dfXSize > INT_MAX || dfYSize > INT_MAX)
+    {
+        dfXSize /= 2;
+        dfYSize /= 2;
+    }
+
+    nRasterXSize = std::max(1, static_cast<int>(0 + 5 + dfXSize));
```

Comment:
Not introduced by you, but just spotted this error:
```suggestion
    nRasterXSize = std::max(1, static_cast<int>(0.5 + dfXSize));
```